lesson2/index.js

```diff
@@ -0,0 +1,19 @@
+const ul = document.getElementById("js-target");
+
+const fragment = document.createDocumentFragment();
+const a = document.createElement("a");
+a.href = "1.html";
+
+const img = document.createElement("img");
+img.src = "bookmark.png";
+img.alt = "ブックマーク";
+const text = document.createTextNode("これです");
+
+a.appendChild(img);
+a.appendChild(text);
+
+const li = document.createElement("li");
+li.appendChild(a);
```

Comment:
IMO
You can use 'insertAdjacentElement' to more compact :)

```suggestion
const li = document.createElement('li');
const anchor = document.createElement('a');
const img = document.createElement('img');

anchor.textContent = 'これです';
anchor.href = '1.html';
img.src = 'bookmark.png';
img.alt = 'ブックマーク';

fragment.appendChild(li).appendChild(anchor).insertAdjacentElement('afterbegin', img);
```

Reply:
I learned a lot! Thank you.